batch/batch/worker/worker.py

```diff
@@ -1605,56 +1610,52 @@ def __init__(
         }
 
-        self.heap_size = self.memory_in_bytes - self.stack_size
-
-        user_command_string = job_spec['process']['command']
-        assert len(user_command_string) >= 3, user_command_string
-        self.revision = user_command_string[1]
-        self.jar_url = user_command_string[2]
-        classpath = f'{find_spark_home()}/jars/*:/hail-jars/{self.revision}.jar:/log4j.properties'
+        self.user_command_string = job_spec['process']['command']
+        assert len(self.user_command_string) >= 3, self.user_command_string
+        self.revision = self.user_command_string[1]
+        self.jar_url = self.user_command_string[2]
+        # self.classpath = f'{find_spark_home()}/jars/*:/hail-jars/{self.revision}.jar:/log4j.properties'
 
-        self.command_string = [
-            'java',
-            '-classpath',
-            classpath,
-            f'-Xmx{self.heap_size}',
-            f'-Xss{self.stack_size}',
-            *user_command_string,
-        ]
-
-        self.process = None
         self.deleted = False
         self.timings = Timings(lambda: self.deleted)
         self.state = 'pending'
-        self.logbuffer = bytearray()
+        self.log: Optional[str] = None
+
+        self.jvm = None
+        self.jvm_name = None
 
     def step(self, name):
         return self.timings.step(name)
 
-    async def pipe_to_log(self, strm: asyncio.StreamReader):
-        while not strm.at_eof():
-            self.logbuffer.extend(await strm.readline())
+    def verify_is_acceptable_query_jar_url(self, url: str):
+        n = len(ACCEPTABLE_QUERY_JAR_URL_PREFIX)
+        if url[:n] != ACCEPTABLE_QUERY_JAR_URL_PREFIX:
+            log.error(f'user submitted unacceptable JAR url: {url} for {self}. {ACCEPTABLE_QUERY_JAR_URL_PREFIX}')
+            raise ValueError(f'unacceptable JAR url: {url}')
 
     async def run(self):
-        async with worker.cpu_sem(self.cpu_in_mcpu):
+        async with self.worker.cpu_sem(self.cpu_in_mcpu):
             self.start_time = time_msecs()
 
             try:
+                with self.step('connecting_to_jvm'):
+                    self.jvm = await self.worker.borrow_jvm()
+                    self.jvm_name = str(self.jvm)
+                    self.scratch = self.jvm.root_dir
+
                 self.task_manager.ensure_future(self.worker.post_job_started(self))
 
                 log.info(f'{self}: initializing')
                 self.state = 'initializing'
 
-                os.makedirs(f'{self.scratch}/')
-
-                await check_shell_output(f'xfs_quota -x -c "project -s -p {self.scratch} {self.project_id}" /host/')
-                await check_shell_output(
-                    f'xfs_quota -x -c "limit -p bsoft={self.data_disk_storage_in_gib} bhard={self.data_disk_storage_in_gib} {self.project_id}" /host/'
-                )
+                # await check_shell_output(f'xfs_quota -x -c "project -s -p {self.scratch} {self.project_id}" /host/')
```

Comment:
delete. Also should we make storage requests illegal for JVMJobs?

Reply:
xfs_quota is fine to use outside of a container. We set the quota on the host directories and then mount them as /io. There's nothing special. I think the code should just work.

Reply:
Actually, you'll need Java 11, the JVMEntryway.class file, and the junixsocket JAR. I think that plus the aforementioned mounts is sufficient.